perspective_reddit_bot/moderate_subreddit.py

```diff
@@ -0,0 +1,152 @@
+"""A reddit bot which uses the Perpsective API to help moderate subreddits.
+
+Args:
+      subreddit: (required) A positional argument for the name of the subreddit to moderate.
+      -rule_config_file: (optional) The file which contains moderation rules. Defaults to 'rules.yaml'
+      -output_dir: (optional) If set, scores for all streamed commments are written to a file in this directory.
+
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import os
+import time
+
+from creds import creds
+import pandas as pd
+import perspective_client
+from perspective_rule import Rule
+import praw
+import score_dataset
+import yaml
+
+TEXT_COLUMN = 'comment_text'
+
+LANGUAGE = 'en'
+# TODO(nthain): support automated language detection.
+
+
+def parse_rules(filepath):
+  rules = []
+  models = []
+  with open(filepath) as f:
+    raw_rules = yaml.load_all(f)
+    for r in raw_rules:
+      rules.append(Rule(r['perspective_score'],
+                        r['action'],
+                        r.get('report_reason')))
+      models.extend(r['perspective_score'].keys())
```

Comment:
this can contain duplicates if the same model is used in multiple rules. the API handles this fine, i believe, but maybe better to use a set() here and convert to list when returning result?

Reply:
Good catch! Done.